src/liballoc/collections/vec_deque.rs

```diff
@@ -202,6 +202,22 @@ impl<T> VecDeque<T> {
                                  len);
     }
 
+    /// Copies all values from `src` to `self`, wrapping around if needed.
+    /// Assumes capacity is sufficient.
+    #[inline]
+    unsafe fn copy_slice(&mut self, src: &[T]) {
+        let dst_high_ptr = self.ptr().add(self.head);
```

Comment:
This should at least `debug_assert!` the pre-condition that `self.capacity() >= self.len().checked_add(src.len()).expect("overflow")`.